File: InsertionSort/insertion.lua
function insertionSort(arr)
    local n = #arr

    for j = 2, n do
        local key = arr[j]
        local i = j - 1

        while i >= 1 and arr[i] > key do
            arr[i + 1] = arr[i]
            i = i - 1
        end
        arr[i + 1] = key;
    end
end

function main()
    io.write("Enter number of elements: ")
    local n = io.read("*n")
    local arr = {}

    for i = 1, n do
        io.write("Enter an element: ")
        arr[i] = io.read("*n")
    end

    insertionSort(arr)

    print("Sorted array is: ")

    for i = 1, n do
        io.write(arr[i].." ")
    end
end

main()

File: InsertionSort/insertion.py
def insert(arr, j):
    for i in range(j-1, -1, -1):
        if arr[i] > arr[i+1]:
            arr[i], arr[i+1] = arr[i+1], arr[i]
        else:
            break

def insertionSort(arr):
    for i in range(len(arr)):
        insert(arr, i)
    return arr

if __name__ == "__main__":
    nums = list(map(int, input("Enter elements of the array (space separated): ").split()))
    s = insertionSort(nums)
    print(f"The sorted array is: {s}")
